pennylane/ftqc/parametric_midmeasure.py

```diff
@@ -0,0 +1,252 @@
+
+        return _label
+
+
+@_add_operation_to_drawer.register
+def _(op: ParametricMidMeasureMP, drawer, layer, _):
+    text = op.plane
+    drawer.measure(layer, op.wires[0], text=text)  # assume one wire
+
+    if op.reset:
+        drawer.erase_wire(layer, op.wires[0], 1)
+        drawer.box_gate(
+            layer + 1,
+            op.wires[0],
+            "|0⟩",
+            box_options={"zorder": 4},
+            text_options={"zorder": 5},
+        )
+
+
+def null_postprocessing(results):
+    """A postprocessing function returned by a transform that only converts the batch of results
+    into a result for a single ``QuantumTape``.
+    """
+    return results[0]
+
+
+@qml.transform
+def diagonalize_mcms(tape):
+    """Diagonalize any mid-circuit measurements in a parameterized basis into the computational basis.
+
+    Args:
+        tape (QNode or QuantumScript or Callable): The quantum circuit to modify the mid-circuit measurements of.
+
+    Returns:
+        qnode (QNode) or tuple[List[QuantumScript], function]: The transformed circuit as described in :func:`qml.transform <pennylane.transform>`.
+
+    **Examples:**
+
+    This transform allows us to transform mid-circuit measurements into the measurement basis by adding
+    the relevant diagonalizing gates to the tape just before the measurement is performed.
+
+    .. code-block:: python3
+
+        from pennylane.ftqc import diagonalize_mcms, ParametricMidMeasureMP
+
+        dev = qml.device("default.qubit")
+
+        @diagonalize_mcms
+        @qml.qnode(dev)
+        def circuit(x):
+            qml.RY(x[0], wires=0)
+            ParametricMidMeasureMP(0, angle=x[1], plane="XY")
+            return qml.expval(qml.Z(0))
+
+    Applying the transform inserts the relevant gates before the measurement to allow
+    measurements to be in the Z basis, so the original circuit
+
+    >>> print(qml.draw(circuit, level=0)([np.pi/4, np.pi/4]))
+    0: ──RY(0.79)──┤↗ˣʸ(3.14)├─┤  <Z>
+
+    becomes
+
+    >>> print(qml.draw(circuit)([np.pi/4, np.pi/4]))
+    0: ──RY(0.79)──U(M0)──┤↗ᶻʸ(0.0)├─┤  <Z>
+    M0 =
+    [[ 0.70710678+0.j   0.5       -0.5j]
+     [ 0.70710678+0.j  -0.5       +0.5j]]
+
+    """
+
+    new_operations = []
+
+    for op in tape.operations:
+        # ToDo: maybe this in-place modification is a bad idea, but
+        #  its also weird to diagonalize it and not update it
+        if isinstance(op, ParametricMidMeasureMP):
+            diag_gate = op.diagonalizing_gates()[0]
+            op.angle = 0
+            op.plane = "ZY"
+            new_operations.extend([diag_gate, op])
+        elif isinstance(op, qml.ops.Conditional) and isinstance(op.base, ParametricMidMeasureMP):
+            diag_gate = qml.ops.Conditional(
+                expr=op.hyperparameters["meas_val"], then_op=op.diagonalizing_gates()[0]
+            )
+            op.base.angle = 0
+            op.base.plane = "ZY"
+            new_operations.extend([diag_gate, op])
```

Comment:
Instead of doing this, you could just keep a list of all P-MCMs and cond(P-MCM)s, and replace them with new ones, where the `meas_val` is updated to use copies of the original P-MCMs. Let me know if that doesn't make sense, I think I might be able to explain better in-person.

Reply:
>Instead of doing this, you could just keep a list of all P-MCMs and cond(P-MCM)s, and replace them with new ones, where the meas_val is updated to use copies of the original P-MCMs. 

That does sound tidier, let me try that out.

> Also, do we not also need to append adjoint(diag_gate) after the measurement?

For MBQC purposes, we are done with a qubit once it's measured (possibly with the exception of reseting it), so we don't care if it ends up back measurement plane; it would just be extra gates. But maybe its worth adding a warning on the transform about this?

Reply:
Updated the transform as suggested :)